test/jdk/javax/swing/JCheckBox/4449413/bug4449413.java

```diff
@@ -25,87 +25,170 @@
-                        }
-                    });
-
-                    getContentPane().add(panel);
-                    panel.setLayout(new GridLayout(4, 6, 10, 15));
-                    for (int k = 0; k <= 3; k++) {
-                        for (int j = 1; j >= 0; j--) {
-                            AbstractButton b = createButton(j, k);
-                            panel.add(b);
-                        }
-                    }
+        add(testedPanel);
+
+
+        if (isMetalLookAndFeel()) {
+            JCheckBox oceanThemeSwitch = new JCheckBox("Use Ocean theme", true);
+            oceanThemeSwitch.addItemListener(e -> {
+                if (e.getStateChange() == ItemEvent.SELECTED) {
+                    MetalLookAndFeel.setCurrentTheme(oceanTheme);
+                } else {
+                    MetalLookAndFeel.setCurrentTheme(defaultMetalTheme);
                 }
+                SwingUtilities.updateComponentTreeUI(testedPanel);
             });
 
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+            add(oceanThemeSwitch);
         }
+
+        JTextArea instructionArea = new JTextArea(
+                isMetalLookAndFeel()
+                        ? INSTRUCTIONS + INSTRUCTIONS_ADDITIONS_METAL
+                        : INSTRUCTIONS
+                );
+
+        instructionArea.setEditable(false);
+        instructionArea.setFocusable(false);
+        instructionArea.setMargin(new Insets(10,10,10,10));
+
+        add(instructionArea);
+
+
+        JButton passButton = new JButton("Pass");
+        JButton failButton = new JButton("Fail");
+
+        ActionListener actionListener = e -> {
+            failed = e.getSource() == failButton;
+            latch.countDown();
+        };
+
+        passButton.addActionListener(actionListener);
+        failButton.addActionListener(actionListener);
+
+        JPanel passFailPanel = new JPanel();
+        passFailPanel.add(passButton);
+        passFailPanel.add(failButton);
+
+        add(passFailPanel);
     }
 
     static AbstractButton createButton(int enabled, int type) {
-        AbstractButton b = null;
-        switch (type) {
-            case 0:
-                b = new JRadioButton("RadioButton");
-                break;
-            case 1:
-                b = new JCheckBox("CheckBox");
-                break;
-            case 2:
-                b = new JRadioButtonMenuItem("RBMenuItem");
-                break;
-            case 3:
-                b = new JCheckBoxMenuItem("CBMenuItem");
-                break;
-        }
+        AbstractButton b = switch (type) {
+            case 0 -> new JRadioButton("RadioButton");
+            case 1 -> new JCheckBox("CheckBox");
+            case 2 -> new JRadioButtonMenuItem("RBMenuItem");
+            case 3 -> new JCheckBoxMenuItem("CBMenuItem");
+            default -> throw new IllegalArgumentException("type should be in range of 0..3");
+        };
+
+        b.setOpaque(true);
```

Comment:
Any reason why this setOpaque was added? I am of the opinion that it should get controlled either by LookAndFeel.installProperty or by component's constructor.

Reply:
The reason is in `JComponent#setBackground()` [javadoc](https://docs.oracle.com/en/java/javase/15/docs/api/java.desktop/javax/swing/JComponent.html#setBackground(java.awt.Color))



> Sets the background color of this component. The background color is used only if the component is opaque, and only by subclasses of JComponent or ComponentUI implementations. Direct subclasses of JComponent must override paintComponent to honor this property.
> 
> It is up to the look and feel to honor this property, some may choose to ignore it.